### PR TITLE
upgrade babel and eslint

### LINKED ISSUE #6496: Update eslint/babel
We need to update these together it appears to appease eslint requirements. See the errors on these two renovate PRs:

- https://github.com/internetarchive/openlibrary/pull/6495
- https://github.com/internetarchive/openlibrary/pull/6494

### Stakeholders
@tangym27 @jimman2003

### PR DESCRIPTION
**Pull request by RayBB**
Closes #6496

Getting us out of dependency hell in preparation for #8309 which is in prep for #8257 .

Lets merge this soon and be done with this headache.

### Technical

The librarian change is due to:
https://eslint.vuejs.org/rules/no-use-computed-property-like-method.html

### Testing

Extensive testing should be done.

My local env is being kind of wonky but it is working great on gitpod
https://gitpod.io/#https://github.com/internetarchive/openlibrary/pull/8312

### Stakeholders

@cclauss have any interest in helping test and get this through the finish line?

### CODE REVIEW

**Comment by cclauss** [comment]
It is great to see progress on this.  Thanks for doing it.  I am unqualified to review JavaScript-related work but I appreciate the modernization effort.

**Comment by jimman2003** [comment]
Hmm, looking on the commits I see that eslint is handled by vue-cli, maybe we should update all of vue-cli together?

**Comment by RayBB** [comment]
@jimman2003 I'm trying to keep the scope of this PR small.
Normally I would say we should update all of them together. But right now when I upgrade the vue-cli it starts using webpack 5 and more things break. So I am trying to do the minimum that I can upgrade here so the PR updating Vue will just focused on Vue/webpack.

File: openlibrary/components/MergeUI.vue [hunk 1]
```diff
@@ -105,7 +105,7 @@ export default {
                     }
                 } catch (e) {
                     this.mergeOutput = e.message;
-                    this.mergeStatus = this.isSuperLibrarian() ? DO_MERGE : REQUEST_MERGE;
+                    this.mergeStatus = this.isSuperLibrarian ? DO_MERGE : REQUEST_MERGE;
```
Comment (jimchamp) [inline openlibrary/components/MergeUI.vue:108]: This change may be unrelated to this PR, and was already merged into main as a part of #8259.

**Comment by cclauss** [comment]
Rubber Stamp Looks Good To Me (RSLGTM)

**Review by jimman2003 — APPROVED** [review]
I tested locally , building works

**Pull request merged**